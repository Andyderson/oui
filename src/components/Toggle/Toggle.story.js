import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';

import Toggle from './index.js';

const stories = storiesOf('Toggle', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories.add('default', withInfo()(() => {
  return <Toggle />;
}));


