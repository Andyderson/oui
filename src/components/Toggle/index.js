import React from 'react';
import PropTypes from 'prop-types';

const Toggle = ({ onClick }) => {
  return (
    <label className="oui-switch" onClick={onClick}>
      <input type="checkbox" />
      <span className="slider round"></span>
      <span className="slider-text"></span>
    </label>
  );
};

Toggle.propTypes = {
  // checked true means the toggle is on off means off
  checked: PropTypes.bool,
  // onClick function
  onClick: PropTypes.func,
};

export default Toggle;

