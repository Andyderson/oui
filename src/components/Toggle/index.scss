////
/// Toggle
/// @description switch a state to on/off. Target `.#{$namespace}overlay` to position
/// as needed.
///
/// @example[html] Default spinner
///   <div class="position--relative height--100">
///     <div class="#{$namespace}overlay">
///       <div class="#{$namespace}spinner"></div>
///     </div>
//    </div>


/* The switch - the box around the slider */
.oui-switch {
  position: relative;
  display: inline-block;
  width: 60px;
  height: 34px;

  input {display:none;}

  input:checked + .slider {
    background-color: #00CC00;
  }
  
  input:focus + .slider {
    box-shadow: 0 0 1px #2196F3;
  }
  
  input:checked + .slider:before {
    -webkit-transform: translateX(26px);
    -ms-transform: translateX(26px);
    transform: translateX(26px);
  }
  
  .slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: #ccc;
    -webkit-transition: .4s;
    transition: .4s;
  }

  .slider-text {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    -webkit-transition: .4s;
    transition: .4s;
  }

  .slider.round {
    border-radius: 34px;
  }
  
  .slider.round:before {
    border-radius: 50%;
  }

  .slider-text:after {
    content: 'Off';
    position: absolute;
    left: 34px;
    top: 9px;
    color: white;
    font-family: Arial;
    font-size: 13px;
  }
  
  .slider-text:before {
    content: 'On';
    position: absolute;
    left: 8px;
    top: 9px;
    color: white;
    font-family: Arial;
    font-size: 13px;
  }
  
  .slider:before {
    position: absolute;
    content: "";
    height: 26px;
    width: 26px;
    left: 4px;
    bottom: 4px;
    background-color: white;
    -webkit-transition: .4s;
    transition: .4s;
  }
}
