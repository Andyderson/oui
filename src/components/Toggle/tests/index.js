import React from 'react';
import Toggle from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('components/Toggle', () => {
  let component;

  beforeEach(() => {
    component = shallow(
      <Toggle />
    );
  });

  it('should render an `Toggle` with correct props', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render an `Toggle` with correct props', () => {
    expect(component.length).toBe(1);
  });

});
