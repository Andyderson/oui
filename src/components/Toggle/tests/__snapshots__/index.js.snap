// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Toggle should render an \`Toggle\` with correct props 1`] = `
<label
  className="oui-switch"
>
  <input
    type="checkbox"
  />
  <span
    className="oui-switch-slider"
  />
  <span
    className="oui-switch--text"
  />
</label>
`;
